model/range_util.sail

```diff
@@ -0,0 +1,33 @@
+/*=======================================================================================*/
+/*  This Sail RISC-V architecture model, comprising all files and                        */
+/*  directories except where otherwise noted is subject the BSD                          */
+/*  two-clause license in the LICENSE file.                                              */
+/*                                                                                       */
+/*  SPDX-License-Identifier: BSD-2-Clause                                                */
+/*=======================================================================================*/
+
+// Return true if the right-open range `a` is a subset of `b` (or equal to it).
+// Ranges wrap around the end. The 64 bit limit is just so Sail generates more
+// efficient C code.
+function range_subset forall 'n, 0 <= 'n <= 64 . (
+  a_begin : bits('n),
+  a_end   : bits('n),
+  b_begin : bits('n),
+  b_end   : bits('n),
+) -> bool = {
```

Comment:
Since the usage of this is with `(addr, width)` tuples, it might be better to have that as the API instead of `[begin, end)`?   It would also localize checking for overflows.  Otherwise this function would need to somehow handle for e.g. `a_begin > a_end` if it ends up having other users.

Reply:
Done.